docs/reference/datastar-sdk/index.md

```diff
@@ -0,0 +1,627 @@
+
+Each of the attributes may have a couple of modifiers, about which you can learn in their respective sections in the [Datastar documentation](https://data-star.dev/reference/attributes). They are all modeled as a type-safe DSL in the `zio-http-datastar-sdk` module. For example, you can use the `debounce` modifier on the `dataOn.input` attribute to debounce input events:
+
+```scala mdoc:compile-only
+import zio._
+
+dataOn.input.debounce(300.millis) := Js("@get('/search?q=' + $query)")
+// datastar equivalent: <p data-on-input__debounce.300ms="@get('/search?q=' + $query)"></p>
+```
+
+Another important attribute is `dataSignals`, which allows you to declare signals that can be used in the Datastar expressions. You can declare a signal using the `dataSignals` attribute as follows:
+
+```scala mdoc:compile-only
+dataSignals(Signal[String]("currentTime")) := js"00:00:00"
+```
+
+This declares a signal named `currentTime` of type `String` with an initial value of `00:00:00`. You can then use this signal in other Datastar attributes. For example, you can use `dataText` to display the current time:
+
+```scala mdoc:compile-only
+span(
+   dataSignals(Signal[String]("currentTime")) := js"",
+   dataText := Js("$currentTime"),
+   dataOn.load := Js("@get('/server-time')"),
+)
+```
+
+In this example, the `dataText` attribute binds the text content of the `span` element to the `currentTime` signal, and the `dataOn.load` attribute triggers a server request to update the signal when the page loads. 
+
+We will discuss later how the server sends updates to the signals using the [`ServerSentEventGenerator#patchSignals`](#patching-signals) method.
+
+## Extracting Datastar Signals from Requests
+
+When the client sends a request to the server, it includes the current values of all signals in a special query parameter named `datastar`. You can extract these signals from the request.
+
+For example, the following HTML form binds an input field to a signal named `delay`; when the form is submitted, the current value of the `delay` signal is sent to the server:
+
+```scala mdoc:compile-only
+body(
+  h1("Hello World Example"),
+  div(
+    dataSignals(Signal[Int]("delay")) := js"100",
+    label("Delay (ms): ", `for` := "delay"),
+    input(`type` := "number", step := "100", dataBind("delay")),
+  ),
+  button(dataOn.click := Js("@get('/hello-world')"))("Start Animation"),
+  div(id("message"))
+)
+```
+
+The server can extract the signals from the request like this:  
+
+```scala mdoc:silent
+import zio._
+import zio.json._
+import zio.http._
+import zio.http.datastar._
+
+case class Delay(value: Int)
+object Delay {
+  implicit val jsonCodec: JsonCodec[Delay] = DeriveJsonCodec.gen
+}
+
+val route =
+  Method.GET / "hello-world" -> events {
```

Comment:
I think we should in general push the endpoint API more. I would like to have the basic Datastar examples with Endpoint API instead.